dart/lib/src/sentry.dart

```diff
@@ -21,11 +28,19 @@ class Sentry {
   static Hub get currentHub => _hub;
 
   /// Initializes the SDK
-  static Future<void> init(OptionsConfiguration optionsConfiguration) async {
+  static Future<void> init(
+    OptionsConfiguration optionsConfiguration,
+    AppRunner callback,
+  ) async {
     if (optionsConfiguration == null) {
       throw ArgumentError('OptionsConfiguration is required.');
     }
+    if (callback == null) {
+      throw ArgumentError('AppRunner is required.');
+    }
```

Comment:
what's about making this an optional parameter for the Dart SDK? not sure if it makes sense to every Dart App.
and you only add the `runZonedGuardedIntegration` if the `callback` is not null

Reply:
should a user make the decision between using `runZonedGuardedIntegration` or not ?

Reply:
it's rather a question because I dont know all the use cases.
do you think that all the devs would like the `runZonedGuardedIntegration` or some people prefer to opt out? we can keep like this now, but it'd be nice to have an idea

Reply:
I'm convinced that it should be an optional param for the Dart SDK, as a Console App or any other type of App that don't have an UI won't make much sense to execute a callback, they don't need to run anything inside of a ZonedGuarded

Reply:
✅ I made it optional.